### DISCUSSION TITLE
How to display workspace on 1 screen (primary monitor)

### DISCUSSION THREAD

**Discussion by s-edupuganti** [i3/i3#5052] [Q&A]
I currently have a laptop connected to an external monitor via USB-C to HDMI, and this is my first time messing around with i3. I usually don't prefer to extend displays since I would like to use my monitor as the main screen. On Windows, you are able to display on one screen, which in this case, is the monitor, but I was wondering how to go about doing this with i3? 

Not sure if this makes a difference, but the laptop I have is a HIDPI display, while the monitor is a regular 1080p monitor that supports 144hz. Basically, when I connect the cable, I want to use the monitor (while laptop screen is not on), while still getting audio and camera from laptop, but when I disconnect the cable, everything will show back up on the laptop.

**Comment by dwedia** [comment]
You can set up xrandr scripts for each screen setup you want (easily done with arandr :) ), and then launch then with specific keybinds.

I have two scripts I use for this

at_work.sh  (This script sets my 2 external monitors up, and shuts off my laptop display).
#!/bin/sh
xrandr --output eDP --off --output HDMI-A-0 --off --output DisplayPort-0 --off --output DisplayPort-1 --off --output DisplayPort-2 --primary --mode 3440x1440 --pos 1440x791 --rotate normal --output DisplayPort-3 --off --output DisplayPort-4 --mode 2560x1440 --pos 0x0 --rotate left

laptop.sh  (This script disables all external displays, and turns on my laptop display)
#!/bin/sh
xrandr --output eDP --primary --mode 1920x1080 --pos 0x0 --rotate normal --output HDMI-A-0 --off --output DisplayPort-0 --off --output DisplayPort-1 --off --output DisplayPort-2 --off --output DisplayPort-3 --off --output DisplayPort-4 --off

I have then bound each script to a hotkey in my i3 config
#change to laptop screen setup
bindsym $mod+Shift+o exec ~/Scripts/ScreenLayout/laptop.sh

#change to work screens setup
bindsym $mod+Shift+p exec ~/Scripts/ScreenLayout/at_work.sh
